pkg/blockstorage/azure/auth.go

```diff
@@ -55,12 +56,26 @@ type AzureAuthenticator interface {
 	GetAuthorizer() azcore.TokenCredential
 }
 
-func NewAzureAuthenticator(config map[string]string) (AzureAuthenticator, error) {
+type NewManagedIdentityCredentialFunc func(opts *azidentity.ManagedIdentityCredentialOptions) (*azidentity.ManagedIdentityCredential, error)
+type NewClientSecretCredentialFunc func(tenantID, clientID, clientSecret string, opts *azidentity.ClientSecretCredentialOptions) (*azidentity.ClientSecretCredential, error)
+type AzIdentityType struct {
+	NewManagedIdentityCredential NewManagedIdentityCredentialFunc
+	NewClientSecretCredential    NewClientSecretCredentialFunc
+}
+
+func NewAzureAuthenticator(config map[string]string, identity *AzIdentityType) (AzureAuthenticator, error) {
```

Comment:
This function is only called with `nil` as a second argument.

Reply:
It is overridden in unit tests